Demo加了moshi1.13.0后编译不过 javassist.NotFoundException

Demo插件里加了这几行代码
sample-app 
        gradle.build 
                apply plugin: 'kotlin-android'
                apply plugin: 'kotlin-kapt'
                api 'com.squareup.moshi:moshi:1.13.0'
                kapt 'com.squareup.moshi:moshi-kotlin-codegen:1.13.0'
rootProject
        gradle.properties
                android.jetifier.ignorelist=moshi-1.13.0
运行host
javassist.NotFoundException: com.squareup.moshi.RecordJsonAdapter$ComponentBinding
Caused by: javassist.NotFoundException: com.squareup.moshi.RecordJsonAdapter$ComponentBinding
	at javassist.ClassPool.get(ClassPool.java:430)
	at com.tencent.shadow.core.transform_kit.ReplaceClassName.checkAll(ReplaceClassName.kt:51)
	at com.tencent.shadow.core.transform_kit.AbstractTransform.checkReplacedClassHaveRightMethods(AbstractTransform.kt:147)
	at com.tencent.shadow.core.transform_kit.AbstractTransform.onCheckTransformedClasses(AbstractTransform.kt:110)
	at com.tencent.shadow.core.transform_kit.AbstractTransform.afterTransform(AbstractTransform.kt:76)
	at com.tencent.shadow.core.transform_kit.ClassTransform.transform(ClassTransform.kt:177)
	at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:284)
	at com.android.build.gradle.internal.profile.NoOpAnalyticsService.recordBlock(NoOpAnalyticsService.kt:72)
	at com.android.build.gradle.internal.pipeline.TransformTask.transform(TransformTask.java:242)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
	... 124 more

尝试ReplaceClassName.checkAll方法循环里 try...catch 然后各种dex报错...，大佬帮忙看看是啥原因导致的

--- a/projects/sdk/core/transform-kit/src/main/kotlin/com/tencent/shadow/core/transform_kit/AbstractTransform.kt
+++ b/projects/sdk/core/transform-kit/src/main/kotlin/com/tencent/shadow/core/transform_kit/AbstractTransform.kt
@@ -23,7 +23,6 @@ import javassist.ClassPool
 import javassist.CtClass
 import org.gradle.api.Project
 import java.io.*
-import java.io.File
 import java.util.zip.ZipEntry
 import java.util.zip.ZipOutputStream
 import kotlin.system.measureTimeMillis
@@ -82,14 +81,6 @@ abstract class AbstractTransform(
     }
 
     private fun CtClass.debugWriteJar(outputEntryName: String?, outputStream: ZipOutputStream) {
-        //忽略META-INF
-        if (outputEntryName != null
-            && listOf<(String) -> Boolean>(
-                { it.startsWith("META-INF/") },
-                { it == "module-info.class" },
-            ).any { it(outputEntryName) }
-        ) return
-
         try {
             val entryName = outputEntryName ?: (name.replace('.', '/') + ".class")
             outputStream.putNextEntry(ZipEntry(entryName))

--- a/projects/sdk/core/transform-kit/src/main/kotlin/com/tencent/shadow/core/transform_kit/ClassTransform.kt
+++ b/projects/sdk/core/transform-kit/src/main/kotlin/com/tencent/shadow/core/transform_kit/ClassTransform.kt
@@ -90,15 +90,25 @@ abstract class ClassTransform(val project: Project) : Transform() {
             it.jarInputs.forEach {
                 val transformInput = TransformInput(it)
                 inputSet.add(transformInput)
-                ZipInputStream(FileInputStream(it.file)).use {
-                    var entry: ZipEntry? = it.nextEntry
-                    while (entry != null) {
-                        if (!entry.isDirectory && entry.name.endsWith(SdkConstants.DOT_CLASS)) {
-                            val inputClass = JarInputClass()
-                            inputClass.onInputClass(it, entry.name)
-                            transformInput.addInputClass(inputClass)
-                        }
-                        entry = it.nextEntry
+                ZipInputStream(FileInputStream(it.file)).use { zis ->
+                    var entry: ZipEntry?
+                    while (true) {
+                        entry = zis.nextEntry
+                        if (entry == null) break
+
+                        val name = entry.name
+
+                        // 忽略一些实际上不会进入编译classpath的文件
+                        if (entry.isDirectory) continue
+                        if (!name.endsWith(SdkConstants.DOT_CLASS)) continue
+                        if (name.startsWith("META-INF/", true)) continue
+                        if (name.endsWith("module-info.class", true)) continue
+                        if (name.endsWith("package-info.class", true)) continue
+
+                        // 记录好entry和name的关系，添加再添加成transform的输入
+                        val inputClass = JarInputClass()
+                        inputClass.onInputClass(zis, name)
+                        transformInput.addInputClass(inputClass)
                     }
                 }
             }

